### PR TITLE
Add docs

### PR DESCRIPTION
**Pull request by hamelsmu**
Three things I added docs about:

1. Confusion that can happen around tokenization
2. How to add special tokens
3. Clarification of how to correctly apply the llama format

### CODE REVIEW

**Comment by NanoCode012** [comment]
Please rebase. I think you don't have latest readme changes. 

Ex: Mistral MOE gone from readme, info about latest eval_per_epoch missing

**Comment by hamelsmu** [comment]
@NanoCode012 Thanks.  Should be good now

**Comment by hamelsmu** [comment]
~Ahh nevermind I thought it was just the linter that was changing the table but still off~ Fixed it

File: README.md [hunk 1]
```diff
@@ -1048,6 +1073,20 @@ It's safe to ignore it.
...
 See the [NCCL](docs/nccl.md) guide.
 
+
+### Tokenization Mismatch b/w Inference & Training
+
+For many formats, Axolotl constructs prompts by concatenating token ids _after_ tokenizing strings.  The reason for concatenating token ids rather than operating on strings is to maintain precise accounting for attention masks.
+
+If you decode a prompt constructed by axolotl, you might see spaces between tokens (or lack thereof) that you do not expect, especially around delimiters and special tokens.  When you are starting out with a new format, you should always do the following:
+
+1. Materialize some data using `python -m axolotl.cli.preprocess your_config.yml`, and then decode the first few rows with your model's tokenizer.
```
Comment (winglian) [inline README.md]:
```suggestion
1. Materialize some data using `python -m axolotl.cli.preprocess your_config.yml --debug`, and then decode the first few rows with your model's tokenizer.
```
debug will print one random row out to the CLI in a helpful manner

**Review by winglian — APPROVED** [review]
thank you!

**Pull request merged**